pennylane/ops/op_math/exp.py

```diff
@@ -227,20 +247,24 @@ def decomposition(self):
         return [qml.PauliRot(new_coeff, string_base, wires=self.wires)]
 
     def matrix(self, wire_order=None):
-
-        coeff_interface = math.get_interface(self.coeff)
-        if coeff_interface == "autograd" and math.requires_grad(self.coeff):
+        scalar = [self.scalar] if qml.math.ndim(self.scalar) == 0 else self.scalar
+        coeff_interface = math.get_interface(*scalar)
```

Comment:
For a broadcasted `scalar`, this checks the interface for each entry of the argument, right? Isn't it easier to do
```suggestion
        coeff_interface = math.get_interface(scalar)
        scalar = [self.scalar] if qml.math.ndim(self.scalar) == 0 else self.scalar
```

Reply:
The problem is that `math.get_interface` always returns "numpy" when the input is a list (see function docstring).

Reply:
Ah, but we anyways do not want to support broadcasting via lists, so I believe this would not be a major issue, and be consistent with other operations.

Reply:
Yep you're right. I added your suggestion 👍